cqf-fhir-utility/src/main/java/org/opencds/cqf/fhir/utility/visitor/AbstractKnowledgeArtifactVisitor.java

```diff
@@ -32,4 +38,27 @@ protected List<IBaseBackboneElement> findArtifactCommentsToUpdate(
             throw new UnprocessableEntityException("Version not supported");
         }
     }
+
+    protected List<IDomainResource> getComponents(
+            KnowledgeArtifactAdapter adapter, Repository repository, ArrayList<IDomainResource> resourcesToUpdate) {
+        adapter.getRelatedArtifactsOfType("composed-of").stream().forEach(c -> {
+            final var preReleaseReference = KnowledgeArtifactAdapter.getRelatedArtifactReference(c);
+            Optional<KnowledgeArtifactAdapter> maybeArtifact =
+                    VisitorHelper.tryGetLatestVersion(preReleaseReference, repository);
+            if (maybeArtifact.isPresent()) {
+                if (resourcesToUpdate.stream()
+                        .filter(rtu ->
+                                rtu.getId().equals(maybeArtifact.get().getId().toString())
+                                        && (rtu.getExtension().stream()
+                                                .anyMatch(ext -> ext.getUrl().equals(isOwnedUrl))))
+                        .collect(Collectors.toList())
+                        .isEmpty()) {
```

Comment:
maybe `anyMatch` instead of filter here?
```suggestion
                if (!resourcesToUpdate.stream()
                        .anyMatch(rtu ->
                                rtu.getId().equals(maybeArtifact.get().getId().toString())
                                        && (rtu.getExtension().stream()
                                                .anyMatch(ext -> ext.getUrl().equals(isOwnedUrl))))) {
```